zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs

```diff
@@ -160,6 +163,25 @@ impl InputSource for TransactionRecordsById {
             }
         }
 
+        if selected.len() < 2 {
+            // since we maxed out the target value with only one note, we have an option to grace a note.
+            // we will rescue the biggest dust note
+            unselected.reverse();
+            if let Some(smallest_note) = unselected
```

Comment:
`smallest_note` is a confusing name.  Is this the largest dust note?   How about `largest_dust_note`, or `dust_note_for_grace_input`?